arithmetization/src/main/java/net/consensys/linea/zktracer/module/mmio/MmioPatterns.java

```diff
@@ -182,17 +182,18 @@ public static Bytes16 excision(
 
   public static void updateTemporaryTargetRam(
       MmuData mmuData, final long targetLimbOffsetToUpdate, final Bytes16 newLimb) {
-    final Bytes bytesPreLimb =
-        Bytes.repeat(
-            (byte) 0,
-            (int)
-                (LLARGE
-                    * targetLimbOffsetToUpdate)); // We won't access the preLimb again, so we don't
-    // care
-    // of its value
-    final Bytes bytesPostLimb =
-        mmuData.targetRamBytes().slice((int) ((targetLimbOffsetToUpdate + 1) * LLARGE));
-
-    mmuData.targetRamBytes(Bytes.concatenate(bytesPreLimb, newLimb, bytesPostLimb));
+
+    int limbStart = (int) (LLARGE * targetLimbOffsetToUpdate);
+    int limbEnd = limbStart + LLARGE;
+    byte[] originalRam = mmuData.targetRamBytes().toArray();
+    int sliceSize = Math.max(0, originalRam.length - limbEnd);
+
+    int size = limbStart + newLimb.size() + sliceSize;
+    byte[] updatedRam = new byte[size];
```

Comment:
Here the updated ram size should be the same as the original one ram

Reply:
What's this method you're modifying is doing : you just want to modify a 16 bytes limb the midldle of your memory so :
- you wipe everything before the limb (you could keep it, but as you don't need it anymore, tiu can put 0s instead)
- you modify your 16 bytes limb
- you keep unchanged everything that's after the unmodified limb


Not sure of what you're doing, I'm on mobile with low network

Reply:
> Here the updated ram size should be the same as the original one ram
In my current implementation, it is the case, but any way, we need to follow this logic. The total size is the sum of the parts' sizes that we're putting together.

Reply:
> Not sure of what you're doing, I'm on mobile with low network

I'm doing the same work, but this implementation is faster as you can see in the flamegraphs